imports/plugins/core/graphql/server/no-meteor/util/applyOffsetPaginationToMongoCursor.js

```diff
@@ -0,0 +1,47 @@
+const DEFAULT_LIMIT = 20;
+
+/**
+ * Inspired by https://www.reindex.io/blog/relay-graphql-pagination-with-mongodb/
+ * @name applyOffsetPaginationToMongoCursor
+ * @method
+ * @memberof GraphQL/ResolverUtilities
+ * @summary Adds `skip` and `limit` to a MongoDB cursor as necessary, based on GraphQL
+ *   `first` and `offset` params
+ * @param {Cursor} cursor MongoDB cursor
+ * @param {Object} args An object with `offset` or `last` but not both.
+ * @param {Object} options Options
+ * @param {Boolean} [options.includeHasNextPage] Whether to return the `pageInfo.hasNextPage`.
+ *   Default is `true`. Set this to `false` if you don't need it to avoid an extra database command.
+ * @return {Promise<Object>} `{ hasNextPage, hasPreviousPage }`
+ */
+export default async function applyPaginationToMongoCursor(cursor, { first, last, offset } = {}, {
+  includeHasNextPage = true
+} = {}) {
+  if (offset && last) throw new Error("Request either `last` or `offset` but not both");
+
+  // Rewind the cursor to start at a zero index
+  cursor.rewind();
+
+  // Enforce a `first: 20` limit if no user-supplied limit, using the DEFAULT_LIMIT
+  const limit = first || DEFAULT_LIMIT;
+
+  let hasNextPage = null;
+
+  const hasPreviousPage = offset > 0;
+
+  if (includeHasNextPage) {
+    const nextCursor = cursor.clone();
+    nextCursor.limit(limit + 1);
+    const nextCursorCount = await nextCursor.count();
```

Comment:
Should be resolved in 62a856e6e

Reply:
@cmbirk The `.count` as you have it will be very slow as `skip` gets bigger and bigger. I still suggest you do the skip `offset + limit`, limit 1, `toArray` approach I have above. Otherwise `.count` is probably pretty fast there, too, since limit will be `1`. The main point is that you shouldn't count the full result set + 1. So it would be:

```js
if (includeHasNextPage) {
  const nextCursor = cursor.clone();
  nextCursor.skip(offset + limit);
  nextCursor.limit(1);
  const nextDoc = await nextCursor.count();
  hasNextPage = !!nextDoc;
}
```

Reply:
@aldeed that should be resolve in cf7b7c940. I've updated the tests as well, but not sure the correct things are getting tested as the result count call is passed what we're expecting.